code/controllers/subsystem/starmap.dm

```diff
@@ -577,3 +529,20 @@ SUBSYSTEM_DEF(starmap)
 	sleep(70) //godspeed (want it to line up with the actual jump animation and such
 	ftl_is_spooling = 0
 	return 1
+
+/datum/controller/subsystem/starmap/proc/get_new_event(var/faction = FALSE)
+	var/event
+	switch(faction)
+		if(FTL_NEUTRAL)
+			event = pickweight(neutral_events)
+		if(FTL_SOLGOV)
+			event = pickweight(solgov_events)
+		if(FTL_SYNDICATE)
+			event = pickweight(syndicate_events)
+		if(FTL_NANOTRASEN)
+			event = pickweight(nanotrasen_events)
+		else
+			event = pickweight(all_events)
+
+	var/datum/ftl_event/new_event = new event
+	return new_event
```

Comment:
I'd recomend you make this just return the type and not an instanced event in case you want to add New args in the future.